src/tag_gc.rs

```diff
@@ -4,54 +4,62 @@ use rustc_data_structures::fx::FxHashSet;
 
 use crate::*;
 
+pub type TagVisitor<'a> = dyn FnMut(Option<AllocId>, Option<BorTag>) + 'a;
```

Comment:
This is not the visitor though, it is the function called by the visitor. I don't have immediate ideas for a better name but this name is potentially confusing.